backend/corpora/lambdas/api/v1/curation/collections/common.py

```diff
@@ -105,10 +105,12 @@ def reshape_dataset_for_curation_api(dataset: dict, preview=False) -> dict:
         else:
             dataset[ontology_element] = []
 
-    if value := dataset.pop("name", None):
-        dataset["title"] = value
-    if value := dataset.pop("is_primary_data", None):
-        dataset["is_primary_data"] = is_primary_data_mapping.get(value)
+    if not preview:  # Add these fields only to full (and not preview) Dataset metadata response
+        dataset["revision_of"] = dataset.pop("original_id", None)
+        dataset["title"] = dataset.pop("name", None)
+        if value := dataset.pop("is_primary_data", None):
+            dataset["is_primary_data"] = is_primary_data_mapping.get(value, [])
+
```

Comment:
@Bento007 pls approve of the restructuring of this entire portion. Tests are passing but I want to make sure I'm not unintentionally screwing this up.

Reply:
This means that if `name` or `origin_id` are not populated in the database, we will return `null` for those fields.

Reply:
correct, which is what we're supposed to do